tee_supplicant/src/tee_supplicant.c

```diff
@@ -150,14 +268,25 @@ static int shm_alloc(const struct device *dev, uint32_t num_params,
 		return TEEC_ERROR_OUT_OF_MEMORY;
 	}
 
-	params[0].c = (uint64_t)addr;
+	shm = k_malloc(sizeof(struct suppl_shm));
+	if (!shm) {
+		k_free(addr);
+		return TEEC_ERROR_OUT_OF_MEMORY;
+	}
+	shm->p = addr;
+	shm->id = (uint64_t)addr;
+	shm->size = size;
+	push_shm(shm);
+
+	params[0].c = shm->id;
```

Comment:
You already have MEMREF_SHM_ID() macro. Why you are not using it here?

Reply:
Fixed